packages/backend/src/app/webhooks/webhook-service.ts

```diff
@@ -81,7 +121,56 @@ function extractHostname(url: string): string | null {
     }
 }
 
-interface CallbackParams {
+const getLatestFlowVersionOrThrow = async (flowId: FlowId, projectId: ProjectId): Promise<FlowVersion> => {
+    const flowVersionId = undefined;
+    const includeArtifacts = false;
+
+    const flowVersion = await flowVersionService.getFlowVersion(
+        projectId,
+        flowId,
+        flowVersionId,
+        includeArtifacts,
+    );
+
+    if (isNil(flowVersion)) {
+        logger.error(`[WebhookService#getLatestFlowVersionOrThrow] error=flow_version_not_found flowId=${flowId} projectId=${projectId}`);
+
+        throw new ActivepiecesError({
+            code: ErrorCode.FLOW_NOT_FOUND,
+            params: {
+                id: flowId,
+            },
+        });
+    }
+
+    return flowVersion;
+};
+
+const getFlowOrThrow = async (flowId: FlowId): Promise<Flow> => {
+    const flow = await flowRepo.findOneBy({ id: flowId });
```

Comment:
I would check if flowId Is undefined because it will return random flow

Reply:
done in 5b177252